src/peft/tuners/lora/config.py

```diff
@@ -119,6 +119,37 @@ def __post_init__(self):
             raise ValueError("router_temperature must be greater than 0.")
 
 
+@dataclass
+class BdLoraConfig:
+    """
+    Configuration for BD-LoRA (Block-Diagonal LoRA).
+
+    Args:
+        lora_a_is_blockdiagonal: Modules where LoRA-A is block-diagonal
+        lora_b_is_blockdiagonal: Modules where LoRA-B is block-diagonal
+        nblocks: Number of blocks in block-diagonal matrices
+    """
+
+    lora_a_is_blockdiagonal: Optional[list[str]] = field(
```

Comment:
The name `lora_a_is_blockdiagonal` would suggest the argument to be a bool. How about something like `target_modules_blockdiagonal_a` or even `target_modules_bd_a` (same for `b`)?